cobc/error.c

```diff
@@ -780,6 +822,61 @@ cb_error_x (cb_tree x, const char *fmt, ...)
 	return ret;
 }
 
+enum cb_warn_val
+cb_error_x (cb_tree x, const char *fmt, ...)
+{
+	va_list ap;
+	const enum cb_warn_opt	opt = cb_warn_ignored_error;
+	const enum cb_warn_val	pref = cb_warn_opt_val[opt];
+	enum cb_warn_val	ret = COBC_WARN_AS_ERROR;
+
+	if (ignore_error && pref == COBC_WARN_DISABLED) {
+		return COBC_WARN_DISABLED;
+	}
+
+	va_start (ap, fmt);
+	ret = cb_error_x_ (x, fmt, ap);
+	va_end (ap);
+
+	return ret;
+}
+
+/**
+ * dispatches the given message as a warning if cb_relaxed_syntax_checks holds,
+ * as an error otherwise
+ */
+enum cb_warn_val
+cb_syntax_check (const char *fmt, ...)
```

Comment:
please use this at other places where `cb_flag_relaxed_syntax_checks` is currently tested - filed.c `duplicate_clause_message` , parser.y and typeck.c on several places.